Simplified/Account.swift

```diff
@@ -6,14 +6,17 @@ private let accountSyncEnabledKey        = "NYPLAccountSyncEnabledKey"
 // MARK: AccountDetails
 // Extra data that gets loaded from an OPDS2AuthenticationDocument,
 @objcMembers final class AccountDetails: NSObject {
-  enum AuthType: String {
+  enum AuthType: String, Codable {
     case basic = "http://opds-spec.org/auth/basic"
     case coppa = "http://librarysimplified.org/terms/authentication/gate/coppa"
     case anonymous = "http://librarysimplified.org/rel/auth/anonymous"
+    case oauthIntermediary = "http://librarysimplified.org/authtype/OAuth-with-intermediary"
     case none
   }
   
-  struct Authentication {
+  @objc(AccountDetailsAuthentication)
+  @objcMembers
+  class Authentication: NSObject, Codable, NSCoding {
```

Comment:
AccountDetailsAuthentication because Authentication struct is nested inside AccountDetails struct. I wasn't able to access this struct otherwise in objc, but I might take a second look into this.

struct AccountDetails {
  struct Authentication {
  }
}

Reply:
I don't think it's possible to use nested types in Objective-C. And I'm afraid of name collision in case there is another `Authentication` type in future. I'd leave it as it currently is.